setup.py

```diff
@@ -194,7 +194,7 @@ def run(self, extensions):
 
     def install_requires():
         # TODO: find a way to install pybind11 and ninja directly.
-        return ['cmake', 'flax']
+        return ['cmake', 'cuda-python', 'flax']
```

Comment:
I'm uneasy about adding this dependency. I think it would be easier to write the CUDA/cuDNN logic in C++ and then expose functions in the pybind module: https://github.com/NVIDIA/TransformerEngine/blob/254764b1b964ca0acca5f0d8d6090877353f1efe/transformer_engine/jax/csrc/extensions.cpp#L49

This approach would also avoid the messiness of manually loading libcublasLt.so.

Reply:
@ksivaman, @timmoon10,
JAX doesn't have any CUDA related API. Not sure if it is intended by Google.
And, `cuda-python` doesn't include cuBLASLt API. Thus, I use `ctypes` to load the API. 

For now, only need 3 CUDA APIs and 1 cuBLASLt API are needed, so I think manually make Python-C binding (at `transformer_engine/jax/csrc/extensions.cpp`) is fine.
I will update the code later.

A reminder that if we need more CUDA APIs in Python layers in the future, and JAX still doesn't have them at that time.
I think `cuda-python` might be a better option.

Reply:
50d9e4afd2787f45caf22d26eb13fd66ba58fbc3 remove all `cuda-python` dependence

Reply:
Hm, the lack of CUDA support in JAX is a good point. Using `cuda-python` might be a good idea in the future if it avoids reimplementing a lot of stuff in `transformer_engine_jax`. That said, we should only do it if the functionality is specific to TE+JAX. If it's general to JAX+CUDA, the logic should be upstreamed to JAX or we should release a standalone convenience library so that non-TE users can benefit. If it's a general TE feature, we should put the logic in `common` so that the other frameworks can benefit.